pkg/occlient/occlient.go

```diff
@@ -681,6 +718,102 @@ func secretKeyName(componentName, baseKeyName string) string {
 	return fmt.Sprintf("COMPONENT_%v_%v", strings.Replace(strings.ToUpper(componentName), "-", "_", -1), strings.ToUpper(baseKeyName))
 }
 
+// getS2ILabelValue returns the requested S2I label value from the passed set of labels attached to builder image
+// and the hard coded possible list(the labels are not uniform across different builder images) of expected labels
+func getS2ILabelValue(labels map[string]string, expectedLabelsSet []string) string {
+	for _, label := range expectedLabelsSet {
+		if retVal, ok := labels[label]; ok {
+			return retVal
+		}
+	}
+	return ""
+}
+
+// GetS2IPathsFromBuilderImg returns script path protocol, S2I scripts path, S2I source or binary expected path, S2I deployment dir and errors(if any) from the passed builder image
+func GetS2IPathsFromBuilderImg(builderImage *imagev1.ImageStreamImage) (string, string, string, string, error) {
+
+	// Define structs for internal un-marshalling of imagestreamimage to extract label from it
+	type ContainerConfig struct {
+		Labels map[string]string `json:"Labels"`
+	}
+	type DockerImageMetaDataRaw struct {
+		ContainerConfig ContainerConfig `json:"ContainerConfig"`
+	}
+
+	var dimdr DockerImageMetaDataRaw
+
+	// The label $S2IScriptsURLLabel needs to be extracted from builderImage#Image#DockerImageMetadata#Raw which is byte array
+	dimdrByteArr := (*builderImage).Image.DockerImageMetadata.Raw
+
+	// Unmarshal the byte array into the struct for ease of access of required fields
+	err := json.Unmarshal(dimdrByteArr, &dimdr)
+	if err != nil {
+		return "", "", "", "", errors.Wrap(err, "unable to bootstrap supervisord")
+	}
+
+	// If by any chance, labels attribute is nil(although ideally not the case for builder images), return
+	if dimdr.ContainerConfig.Labels == nil {
+		glog.V(4).Infof("No Labels found in %+v in builder image %+v", dimdr, builderImage)
+		return "", "", "", "", nil
+	}
+
+	// Extract the label containing S2I scripts URL
+	s2iScriptsURL := dimdr.ContainerConfig.Labels[S2IScriptsURLLabel]
+	s2iSrcOrBinPath := dimdr.ContainerConfig.Labels[S2ISrcOrBinLabel]
+
+	if s2iSrcOrBinPath == "" {
+		// In cases like nodejs builder image, where there is no concept of binary and sources are directly run, use destination as source
+		s2iSrcOrBinPath = getS2ILabelValue(dimdr.ContainerConfig.Labels, S2IDeploymentsDir)
```

Comment:
This is not right. If `io.openshift.s2i.destination` is not set, it should use `/tmp` by default. See https://github.com/openshift/source-to-image/blob/master/docs/builder_image.md#required-image-contents

Reply:
Its seems not to be the case for nodejs however...
I'll confirm once my minishift is back to normal 
I see that my system /var/ has filled up as in :
```
[root@localhost redhat]# du -h -d 1 /var/
4.0K	/var/games
4.0K	/var/adm
4.0K	/var/local
516K	/var/tmp
4.0K	/var/crash
24K	/var/www
4.0K	/var/gopher
4.0K	/var/ftp
4.0K	/var/nis
4.0K	/var/account
21G	/var/lib
43G	/var/cache
12K	/var/kerberos
4.2G	/var/log
452K	/var/spool
8.0K	/var/empty
4.0K	/var/yp
4.0K	/var/preserve
12K	/var/db
4.0K	/var/opt
67G	/var/

```

Reply:
Perhaps you need to do `minishift delete` and create a new VM :).

NodeJS does indeed work

Reply:
I assume that `/tmp` should be the absolute last resort @kadel ?

Reply:
Incorporated now